packages/java/endpoint/src/test/java/dev/hilla/auth/EndpointAccessCheckerTest.java

```diff
@@ -399,4 +417,258 @@ public void test() {
             CurrentInstance.clearAll();
         }
     }
+
+    @Test
+    public void should_fail_When_Endpoint_is_not_annotated() throws Exception {
```

Comment:
It is not super clear what is being tested here. If we are testing that access is checked from  the correct method/class, maybe we do not need to test separately for all annotations and then test naming will become clearer, e.g. 
```
should_pass_When_Endpoint_overridden_method
```
is clearer that it tests how access is checked compared to `should_pass_When_Endpoint_overridden_method_is_AnonymousAllowed` where you don't get the relevance of `AnonymousAllowed` in the test and start wondering if checking is done using different methods/classes for different annotations

Reply:
Well, I definitely agree with the "naming should be improved" part, but in the first two tests it is testing with an Endpoint that is **not** annotated and inherits a method from a super class. In this case, being overridden is not gonna simply grant the access. If I change it to something like "access_granted_when_inherited_method_is_overridden_and_annotated" it can be still vague since annotating with "@DenyAll" doesn't count, however, as it is the only case, we can ignore and simplify since we are not testing the effect of annotations, but the correct target for access checking. All combinations of annotations don't require testing here, as they are tested separately.

Reply:
But then the test should not be about "access granted" but instead of which access control annotation is used?

Reply:
I think we are almost saying the same thing. I just emphasized that something like "should_pass_When_Endpoint_overridden_method" is not clear either and suggested a new one. The unnecessary tests that were re-checking the logic based on which access control annotation is used are removed now.